platform-sdk/swirlds-logging/src/main/java/com/swirlds/logging/ostream/RolloverFileOutputStream.java

```diff
@@ -0,0 +1,308 @@
+ *     long maxFileSize = 1024 * 1024; // 1 MB
+ *     boolean append = true;
+ *     int maxRollover = 5; // Maximum number of rolling files
+ *     String datePattern = "yyyy-MM-dd"; // Date pattern for naming rolled files
+ *     RolloverFileOutputStream outputStream = new RolloverFileOutputStream(logPath, maxFileSize, append, maxRollover, datePattern);
+ * </pre>
+ * <p>
+ * Note: It's important to manage the lifecycle of the output stream properly by calling {@link #flush()} and
+ * {@link #close()} methods when finished using it to ensure proper resource cleanup.
+ */
+public class RolloverFileOutputStream extends OutputStream {
+    private final Path logPath;
+    private final int maxRollover;
+    private final long maxFileSize;
+    private final boolean append;
+    private final String baseName;
+    private final String extension;
+    private final int indexLength;
+    private final RollingManager rollingManager;
+    private int index;
+    private long remainingSize;
+    private FileOutputStream outputStream;
+
+    /**
+     * Creates an {@link RolloverFileOutputStream}. This constructor is meant for internal uses and test.
+     *
+     * @param logPath         path where the logging file is located. Should contain the base dir + the name of the
+     *                        logging file.
+     * @param maxFileSize     maximum size for the file. The limit is checked with best effort
+     * @param append          if true and the file exists, appends the content to the file. if not, the file is rolled.
+     * @param maxRollover     Within a rolling period, how many rolling files are allowed.
+     * @param datePattern     if not null a {@link SizeAndDateRollingManager} is applied if not
+     *                        {@link SizeRollingManager}
+     * @param instantSupplier is used to get the current {@link Instant} instance. if not set and
+     *                        {@link SizeAndDateRollingManager} the behaviour is defaulted to {@link Instant#now()}
+     */
+    RolloverFileOutputStream(
+            final @NonNull Path logPath,
+            final long maxFileSize,
+            final boolean append,
+            final int maxRollover,
+            final @Nullable String datePattern,
+            final @Nullable Supplier<Instant> instantSupplier) {
+        this.logPath = logPath.toAbsolutePath().getParent();
+        this.append = append;
+        this.maxRollover = maxRollover;
+        this.maxFileSize = maxFileSize;
+        String baseFile = logPath.getFileName().toString();
+        final int i = baseFile.lastIndexOf(".");
+        this.baseName = i >= 0 ? baseFile.substring(0, i) : baseFile;
+        this.extension = i >= 0 ? baseFile.substring(i + 1) : "";
+        this.index = 0;
+        this.indexLength = String.valueOf(maxRollover).length();
+
+        if (datePattern == null || datePattern.isEmpty()) {
+            this.rollingManager = new SizeRollingManager();
```

Comment:
Ended up separating RolloverFileOutputStream into 2 subclasses and eliminating the inner manager classes. I prefer composition to inheritance, but in this case was more confusing than helping.
To create them in an easier way, I created a factory class that receives the configuration and creates the right instance according to the received properties

Reply:
Decision was to remove the support for date+size rolling and provide support only for size, given that is the current usage.
In case we need it in the future, we will add support for date rolling again.